File: Common/Entity/NoteService/Note.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Common.Entity.NoteService
{
    public class Note
    {
        [Key]
        public int Id { get; set; }

        public string Name { get; set; }

        public string Text { get; set; }

        [Required]
        public DateTime LastChange { get; set; }

        [Required]
        public int NoteCategoryId { get; set; }
    }
}

File: MyHomeSPA/src/app/notes/models/note.ts
export class Note {

    public id: number;

    public name: string;

    public text: string;

    public lastChange: Date;

    public noteCategoryId: number;

    constructor() {

        this.id = 0;

        this.name='Заметка';

        this.text='Описание';

        this.lastChange = new Date();

        this.noteCategoryId = 0;
    }

    getCopy(): Note {
        let copyNote = new Note();

        copyNote.id = this.id;
        copyNote.name = this.name;
        copyNote.text = this.text;
        copyNote.lastChange = this.lastChange;
        copyNote.noteCategoryId = this.noteCategoryId;

        return copyNote;
    }
}